### DISCUSSION TITLE
Header redirect

### DISCUSSION THREAD

**Discussion by ghost** [GitbookIO/community#66] [Feature Requests]
### What's your feature/idea?

Allow users to set up redirects for headers and other anchor links. 

### What's your desired outcome?

I want the flexibility to reorganize content in a space without breaking links to headers. For example, on page A I have an H1 block. It has content blocks (e.g. paragraphs and tables) nested beneath it. I create a new page B. On it, I add an H1 with the same title and nested content blocks as exists on page A. 

I then want to set up a redirect from page A, H1 to page B, H1 (ideally in the GUI), save that configuration, and then delete H1 from page A. Now, when a user clicks a link (for example a bookmark in a browser or hypertext in that same space) that points to page A, H1 they are instead sent to page B, H1.

### What's the impact of this for you?

- [X] Critical
- [ ] High
- [ ] Medium
- [ ] Low

**Comment by dfinster** [comment]
We are currently planning a large information architecture overhaul, and the ability to control a list of redirects would be very helpful. For our purposes, it could be as simple as a single page with two columns: **From** and **To** URLs that control [HTTP 301 headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301).

**Comment by dfinster** [comment]
We are now experiencing broken links from the documentation structure changes implemented [since my previous comment](https://github.com/orgs/GitbookIO/discussions/66#discussioncomment-5554402). Some inbound links to our docs from blog posts and external sites are now broken. 

**Redirection is a basic necessity for a product as expensive as GitBook. Even WordPress has this capability.** 

Moving pages or changing headings should not break external links and Google searches. We should have the ability to see the server logs that show the 404 errors, and decide where those links should go.

**Comment by addisonschultz** [comment]
@cgundacker1 are you able to achieve this using GitBook's redirects? https://docs.gitbook.com/product-tour/git-sync/content-configuration#redirects
